### PR TITLE
feat: add hilla-react module to 23.3

### PR DESCRIPTION
**Pull request by cromoteca**
Creates a new module: `dev.hilla:hilla-react`. It will be used in Hilla+React applications instead of `dev.hilla:hilla` to avoid import Vaadin Components directly in `package.json`.

This is related to vaadin/start#2108.

Please review the generated dependencies carefully.

### CODE REVIEW

File: hilla-react/versions.json [hunk 1]
```diff
@@ -0,0 +1,329 @@
+{
```
Comment (ZheSun88) [inline hilla-react/versions.json:1]: does hilla have some logic to parse this file?
for example: 
- replacing the {{version}} to a version number
- keep the version inside to be updated

File: versions.json [hunk 2]
```diff
@@ -5,6 +5,12 @@
             "npmName": "@vaadin/bundles"
         }
     },
+    "react": {
+        "react-components": {
+            "jsVersion": "1.3.0-alpha6",
+            "npmName": "@hilla/react-components"
+        }
+    },
```
Comment (cromoteca) [inline versions.json:13]: Is adding this section safe?

Comment (ZheSun88) [reply versions.json:13]: It is fine to add it here. This file will not be deployed anywhere.

File: hilla-react/pom.xml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>dev.hilla</groupId>
+            <artifactId>endpoint</artifactId>
+            <version>${hilla.version}</version>
```
Comment (ZheSun88) [inline hilla-react/pom.xml:40]: you can remove this `${hilla.version}` here, as the version should come from the hilla-bom. 
same with the `engine-runtime` artifacts.

**Comment by ZheSun88** [comment]
Is there a way to test this? 
or have you tested this locally?

File: hilla-react/pom.xml (hunk 3 above)
Comment (cromoteca) [reply hilla-react/pom.xml:40]: I'll add `endpoint` to the bom.

**Comment by cromoteca** [comment]
> Is there a way to test this? or have you tested this locally?

Good question. I tested locally and it's fine (together with https://github.com/vaadin/start/pull/2108). But of course something more robust is needed. Is there anything I can add on TeamCity maybe?

**Comment by manolo** [comment]
There is a module that is activated with the `fusion-hybrid` profile that I think is skipped since some time ago

**Review by manolo — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**